p4c_bm/gen_json.py

```diff
@@ -1199,6 +1211,39 @@ def dump_field_aliases(json_dict, hlir, path_field_aliases):
     json_dict["field_aliases"] = field_aliases
 
 
+def dump_extern_instances(json_dict, hlir):
+    extern_instances = []
+    id_ = 0
+    for name, p4_extern_instance in hlir.p4_extern_instances.items():
+        extern_instance_dict = OrderedDict()
+        extern_instance_dict["name"] = name
+        extern_instance_dict["id"] = id_
+        extern_instance_dict["type"] = p4_extern_instance.extern_type.name
+
+        id_ += 1
+
+        attributes = []
+        for attribute, attr in p4_extern_instance.attributes.items():
+            attr_type = p4_extern_instance.extern_type.attributes[attribute].\
+                        value_type.type_name
+            if attr_type != "bit" and attr_type != "int":
```

Comment:
Hi Antonin,

where exactly to add the pragma? My program is not also covering this code
snippet?

Regards,
Jeferson

2016-09-13 13:14 GMT-04:00 Antonin Bas <notifications@github.com>:

> In p4c_bm/gen_json.py
> <https://github.com/p4lang/p4c-bm/pull/62#discussion_r78603574>:
>
> > +def dump_extern_instances(json_dict, hlir):
> > +    extern_instances = []
> > +    id_ = 0
> > +    for name, p4_extern_instance in hlir.p4_extern_instances.items():
> > +        extern_instance_dict = OrderedDict()
> > +        extern_instance_dict["name"] = name
> > +        extern_instance_dict["id"] = id_
> > +        extern_instance_dict["type"] = p4_extern_instance.extern_type.name
> > +
> > +        id_ += 1
> > +
> > +        attributes = []
> > +        for attribute, attr in p4_extern_instance.attributes.items():
> > +            attr_type = p4_extern_instance.extern_type.attributes[attribute].\
> > +                        value_type.type_name
> > +            if attr_type != "bit" and attr_type != "int":
>
> Could you add a # pragma: no cover here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/p4lang/p4c-bm/pull/62/files/883e31b0ec4a96622382f348530d043ad1099385#r78603574>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AKNH5jXVVBPrHr_V0u8ZB1QKEQaAa10uks5qptnegaJpZM4J5GHs>
> .
>



-- 
Jeferson Santiago da Silva, M.Sc.
<http://br.linkedin.com/in/engjefersonsantiago>
Polytechnique Montréal
Ph.D. Candidate in Computer Engineering
Hardware and Programmable Logic Design Engineer
Phone: +1 (514) 581-1004
Skype: eng.jefersonsantiago
Montreal/QC - Canada
Lattes CV: http://lattes.cnpq.br/9568850258686622